File: java8/src/main/java/folio/codinginterview/domain/AppConstants.java
package folio.codinginterview.domain;

import java.math.BigDecimal;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public final class AppConstants {
    private AppConstants() {}

    public static final BigDecimal CASH_RATE = new BigDecimal("0.05");

    public static final BigDecimal MIN_OPERATION_AMOUNT = new BigDecimal(10000);

    public static final List<StockSymbol> SUPPORTED_SYMBOLS =
            Collections.unmodifiableList(Arrays.asList(StockSymbol.Toyopa, StockSymbol.Somy));

    public static final Portfolio INITIAL_PORTFOLIO = new Portfolio(Arrays.asList(
            new PortfolioItem(StockSymbol.Toyopa, new BigDecimal("0.40")),
            new PortfolioItem(StockSymbol.Somy, new BigDecimal("0.60"))
    ));
}

File: typescript/src/domain/appConstants.ts
import Decimal from "decimal.js";
import { StockSymbol } from "./stockSymbol.js";
import { Portfolio } from "./stock.js";

/** アプリケーション定数。 */
export const AppConstants = {
  cashRate: new Decimal("0.05"),
  minOperationAmount: new Decimal(10000),
  supportedSymbols: [StockSymbol.Toyopa, StockSymbol.Somy] as ReadonlyArray<StockSymbol>,
  initialPortfolio: new Portfolio([
    { symbol: StockSymbol.Toyopa, rate: new Decimal("0.40") },
    { symbol: StockSymbol.Somy, rate: new Decimal("0.60") },
  ]),
};
